File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/NewGuidShouldNotBeUsed.cs
using System;

namespace Tests.Diagnostics
{
    public class Program
    {
        public void DefaultInitalization()
        {
            Guid default_value = default; // Noncompliant
            var default_T_value = default(Guid); // Noncompliant
        }

        public void WithoutArguments()
        {
            var result = new Guid(); // Noncompliant {{Use 'Guid.NewGuid()' or 'Guid.Empty' or add arguments to this Guid instantiation.}}
            //           ^^^^^^^^^^
        }

        public void WithArguments()
        {
            var bytes = new Guid(new byte[0]); // Compliant
        }

        public void Other()
        { 
            var empty = Guid.Empty; // Compliant
            var rnd = Guid.NewGuid(); // Compliant
        }
    }
}

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/NewGuidShouldNotBeUsed.vb
Imports System

Public Class NewGuid

    Public Sub DefaultInitalization()
        Dim asDefault As Guid = Nothing ' Noncompliant
    End Sub

    Public Sub WithoutArguments()

        Dim id As Guid = New Guid() ' Noncompliant {{Use 'Guid.NewGuid()' or 'Guid.Empty' or add arguments to this Guid instantiation.}}
        '                ^^^^^^^^^^
        Dim noBrackets As Guid = New Guid ' Noncompliant

    End Sub

    Public Sub WithArguments()
        Dim asGuid As Guid = New Guid(New Byte() {}) ' Compliant
        Dim asNewGuid As New Guid(New Byte() {}) ' Compliant
    End Sub

    Public Sub Other()
        Dim empty As Guid = Guid.Empty ' Compliant
        Dim rnd As Guid = Guid.NewGuid() ' Compliant
    End Sub

End Class
